### ISSUE TITLE
[PAID] [$250] [Collect] Workspace - User name and avatar disappears when opening profile immediately after invite

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#38925]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.56-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
Precondition:
- User is admin of Collect workspace.

1. Go to staging.new.expensify.com.
2. Go to Profile > Workspaces > Collect workspace.
3. Go to Members.
4. Invite a new member that does not have Expensify account.
5. Immediately after inviting the member, click on the highlighted new user in the list.

## Expected Result:
The user name and avatar will not disappear.

## Actual Result:
The user name disappears and avatar changes to placeholder avatar.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/e207297d-5428-4893-909c-3717d93be0a6

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~011d6aa43dad80012f</li>
        <li>Upwork Job ID: 1778901063514075136</li>
        <li>Last Price Increase: 2024-04-26</li>
<li>Automatic offers: </li>
<ul>
<li>ikevin127 | Reviewer | 0</li>
<li>nkdengineer | Contributor | 0</li>
<li>bernhardoj | Contributor | 102955490</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @strepanier03</details>

**Comment by izarutskaya** [comment]
We think that this bug might be related to #wave-control.

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The user name and avatar disappear if we view the recently added workspace member. We can easily reproduce this by adding the member while offline, viewing the member detail, and go online.

### What is the root cause of that problem?
When we start adding a new member, we generate an optimistic personal detail for that user and clear it when the request is successful.
https://github.com/Expensify/App/blob/b3baa68ac50d1c75e52385b4e60ce581de7baa24/src/libs/actions/Policy.ts#L1130-L1138
https://github.com/Expensify/App/blob/b3baa68ac50d1c75e52385b4e60ce581de7baa24/src/libs/actions/Policy.ts#L1143-L1161

That's why after the request is completed, the name and avatar disappear because we are viewing the optimistic personal detail data that is already gone.

### What changes do you think we should make in order to solve the problem?

We can disable it when the personal detail is still an optimistic one until we receive the correct personal detail from the server. This is what we do in several places too, for example:
https://github.com/Expensify/App/blob/b3baa68ac50d1c75e52385b4e60ce581de7baa24/src/components/ReportWelcomeText.tsx#L143-L153
https://github.com/Expensify/App/blob/1c632fe194cd634d6024965f13d3605043f32ed6/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js#L422-L425

First, add `isOptimisticPersonalDetail: true` to `getNewPersonalDetailsOnyxData`.
https://github.com/Expensify/App/blob/1c632fe194cd634d6024965f13d3605043f32ed6/src/libs/PersonalDetailsUtils.ts#L125-L138

Then, disable it when `isOptimisticPersonalDetail` is true.
https://github.com/Expensify/App/blob/1c632fe194cd634d6024965f13d3605043f32ed6/src/pages/workspace/WorkspaceMembersPage.tsx#L362-L367

This will disable the checkbox too, we can customize to not disable the checkbox, but I don't think that's a good idea separating the checkbox disabled state and the row disabled state. Or we can keep it enable, but do nothing when clicked.

_optionally we can show not found page on member detail page when it's an optimistic personal detail, but I see we don't do that for profile page, so not sure whether we want to do this or not_
### What alternative solutions did you explore? (Optional)
Don't clear the optimistic personal detail, but this would keep stacking unnecessary data every time we add a new member. Also, it's useless for the user to see the outdated optimistic personal detail when the server already sends us the correct data.

_or don't rerender the detail page when personal details change. we can use memo for this case._

**Comment by bernhardoj** [comment]
Just FYI, we don't need to perform the 5th step quickly. We can perform it while offline to reproduce it easier.
1. Go offline
2. Add a new member (any user that doesn't exist in your contact, it can be a registered expensify user or non-registered, doesn't matter)
3. Click the member to open the detail
4. Go online
5. Wait for a few seconds until the apps reconnecting and the name and avatar will disappear

**Comment by strepanier03** [comment]
Hmmm, I'm having a slightly different experience with the avatar, but the username isn't an issue when testing Mac/Chrome and Android mobile web for me.

----

On both here's what I am doing and seeing:

1. Go to staging.new.expensify.com.
2. Go to Profile > Workspaces > Collect workspace.
3. Go to Members.
4. Invite a new member that does not have Expensify account.
5. Immediately after inviting the member, click on the highlighted new user in the list.

Expected Result:
The user name and avatar will not disappear and remain as they appear until the user updates them..

Actual Result:
In the RHP the username and avatar remain as they appear initially, but in the member table the avatar changes to a different default avatar. Pressing back and then clicking on the new member, shows the updated avatar.

https://github.com/Expensify/App/assets/10925636/17efd9f8-27d2-4f97-a3bf-b79485cc9c78

**Comment by bernhardoj** [comment]
@strepanier03 did you test it on a collect workspace? Looking at the left side of the workspace menu, I guess it's a free workspace. The new RHP for workspace member detail (instead of a profile page that is shown in your video) is only available for collect workspace.

**Comment by strepanier03** [comment]
I'll retest, I made a collect workspace but maybe clicked the wrong button afterward and made on the free workspace.

**Comment by strepanier03** [comment]
Running into an issue recreating this. I get a "Hmm ... its not here" in the RHP when clicking on any Collect workspace member.

<img width="1440" alt="image" src="https://github.com/Expensify/App/assets/10925636/b57acc58-5c7b-4852-885b-2e8dd5072265">

I'm going to ask for a hand repro'ing and then tie to a project.

**Comment by strepanier03** [comment]
Not sure what was different on the second collect workspace I made but I was able to repro with a fresh slate.

https://github.com/Expensify/App/assets/10925636/7f8e95b3-cd63-4859-840f-b9da3d09e21c

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The user name disappears and avatar changes to placeholder avatar.

### What is the root cause of that problem?
The problem here happens because when we add a new workspace member, if that member doesn't exist in personal details list yet, we'll create optimistic personal details for them and then clearing the personal details after the request is done, see [here](https://github.com/Expensify/App/blob/b3baa68ac50d1c75e52385b4e60ce581de7baa24/src/libs/actions/Policy.ts#L1161).

This leads to the user seeing user name and avatar disappearing because the optimistic personal details is gone now.

### What changes do you think we should make in order to solve the problem?

We had this same problem with the optimistic report: we create a new optimistic report with a user, but the report with that user already exists, so what we did is we'll navigate to the new report when the API request is successful.

Here's exactly [how we implement it](https://github.com/Expensify/App/blob/main/src/libs/actions/Report.ts#L1139-L1147), we should do the same for the optimistic personal detail in this case
- Do not clear the personal details of the existing users when the API request is successful
- Modify the back-end to `merge` `preexistingAccountID` into the `personalDetailsList` item of the current optimistic personal detail created in front end. For example, if the `optimisticAccountID` created in front-end is `123456`, and the actual `accountID` of the existing user is `654321`, then the backend will return this Onyx data:
```
{
  "key": "personalDetailsList",
  "onyxMethod": "merge",
  "value": {
    "123456": {
      "preexistingAccountID": 654321
  }
}
```
- Now, in the front-end, when we detect that there's the `preexistingAccountID` being merged into `personalDetailsList`, we'll do the following (same as [here for report](https://github.com/Expensify/App/blob/db9e01cc302df7f2b3dd53a99c1082404676aedb/src/libs/actions/Report.ts#L1140-L1145)):
     + Check if the user is on the workspace member URL with optimistic account ID, if yes, navigate the user to the workspace member URL with the `preexistingAccountID`
     + Clear the optimistic personal details
     
This is the same approach as we did with the optimistic report, just that we do for optimistic personal details now. This will make sure the user has a seamless experience when using the product.

### What alternative solutions did you explore? (Optional)
NA

**Comment by strepanier03** [comment]
@ikevin127 - Friendly bump on the proposal [here](https://github.com/Expensify/App/issues/38925#issuecomment-2053605588), thanks!

**Comment by ikevin127** [comment]
@strepanier03 Sorry, didn't get notified 3 days ago.

I will review the mentioned proposal and get back in the next 24h.

Should I consider @bernhardoj's proposal too, since it was the first one on this issue ?

**Comment by ikevin127** [comment]
@nkdengineer @bernhardoj Could you please provide a video result of your best solution in terms of UI ?

**Note:** for the proposal involving BE changes, try to mock the BE response using a timeout, simply for the sake of demonstrating the video result in terms of UI.

I am asking for this so we can make a decision on which proposal to go with since one of them is proposing a strictly FE fix and the other one involves BE changes. I plan to tag the design team once we have the videos and decide if this could be addressed strictly from the FE side or whether to involve somebody from the BE team and have a BE / FE fix.

**Comment by bernhardoj** [comment]
Here is mine

https://github.com/Expensify/App/assets/50919443/fa49774e-83cd-46fe-b64b-b03f41493b00

**Comment by nkdengineer** [comment]
> @nkdengineer @bernhardoj Could you please provide a video result of your best solution in terms of UI ?

Sure, I'll provide this tomorrow since I need to do some mocking of the BE.

**Comment by ikevin127** [comment]
@bernhardoj Thanks for your early proposal. After careful consideration I decided to move on with the other proposal since as per the **expected result** of the issue, that fits best as a solution to our issue.

**Why ?**
To be more specific I'm referring to the behaviour your video from [here](https://github.com/Expensify/App/issues/38925#issuecomment-2059096784) suggested: even if say we would disable (gray-out) the option until the BE response arrives, that looks more like a workaround considering the expected result, and additionally I did not see this behaviour anywhere else in the app when it comes to a call's optimistic -> success transition.

**Comment by ikevin127** [comment]
**Expected result:**
> The user name and avatar will not disappear.

As per the **expected result** of the issue, I think @nkdengineer's proposal fits best as a solution here. The RCA is correct, the proposed solution fixes the issue as suggested by the expected result.

🎀👀🎀 C+ reviewed

**Comment by ikevin127** [comment]
> [!note]
> Considering BE changes are needed as part of the selected [proposal's](https://github.com/Expensify/App/issues/38925#issuecomment-2053605588) solution, I noticed that in the issue https://github.com/Expensify/App/issues/15511 and PR https://github.com/Expensify/App/pull/24320 which did something similar [here](https://github.com/Expensify/App/blob/db9e01cc302df7f2b3dd53a99c1082404676aedb/src/libs/actions/Report.ts#L1136-L1148), @puneetlath handled the BE side, therefore I'd suggest them for handling the BE part here too if they have the capacity.

@techievivek Otherwise, please let us know who could handle the BE side of the proposed solution. Thank you!

**Comment by techievivek** [comment]
Just a quick question so I understand the RCA and then I can handle the BE changes.

> The problem here happens because when we add a new workspace member

Could you please confirm whether this issue occurs for a user who doesn't exist in our database, or if it happens when the user might exist in the database but we don't have their personal details locally? I ask because if the user doesn't exist, are we sending the optimistic accountID in the request? If so, why doesn't it match what we finally receive in the response?

**Comment by techievivek** [comment]
Reproduction step says this.

> Invite a new member that does not have Expensify account.

**Comment by techievivek** [comment]
In report case, it makes sense that the optimistically generated reportID doesn't match the backend reportID as a chat already exists between users but here the RCA is not clear to me.

**Comment by ikevin127** [comment]
> Could you please confirm whether this issue occurs for a user who doesn't exist in our database

As far as my understanding goes, this is correct. It occurs for a user who doesn't exist in our database.

> I ask because if the user doesn't exist, are we sending the optimistic accountID in the request? If so, why doesn't it match what we finally receive in the response?

> In report case, it makes sense that the optimistically generated reportID doesn't match the backend reportID as a chat already exists between users but here the RCA is not clear to me.

@nkdengineer Could you please clarify ? Thank you!

**Comment by nkdengineer** [comment]
@techievivek @ikevin127 

> Could you please confirm whether this issue occurs for a user who doesn't exist in our database, or if it happens when the user might exist in the database but we don't have their personal details locally?

This happens in both these cases

> I ask because if the user doesn't exist, are we sending the optimistic accountID in the request?

No we're not sending optimistic `accountID` in the request in this case, we're only sending the `emails`

>  If so, why doesn't it match what we finally receive in the response?

@techievivek Even if we send the `accountID` to the back-end (tested), the back-end seems not to be using the optimistic accountID that was sent from front-end to create the user. The back-end will always send back a brand new accountID (different from the one optimistically generated from front-end)

**Comment by ikevin127** [comment]
@techievivek Please let us know if https://github.com/Expensify/App/issues/38925#issuecomment-2066111499 makes sense from a BE point of view when it comes to the selected [proposal](https://github.com/Expensify/App/issues/38925#issuecomment-2053605588)'s solution.

**Comment by techievivek** [comment]
Assigned the issue to @nkdengineer, I will start working on the backend changes for this.

**Comment by nkdengineer** [comment]
@techievivek Let me know when the back-end change is complete.

**Comment by techievivek** [comment]
Sorry for the delay. I couldn't get to this last week as I was focused on other high-priority stuff. I will try to prioritize this tomorrow.

**Comment by ikevin127** [comment]
@techievivek Still reproducible for me on latest main (local dev) and staging (v1.4.73-6), see videos below.

<details>
<summary>MacOS: Chrome</summary>

| DEV | STG |
| ------------- | ------------- |
| <video src="https://github.com/Expensify/App/assets/56457735/7f83dff0-a9c5-4f5c-a6d0-a9e79ff3be19"> | <video src="https://github.com/Expensify/App/assets/56457735/0db8b121-88fe-465b-8bdf-971fdf09e4b6"> |

</details>

**Comment by nkdengineer** [comment]
@techievivek Still reproducible for me on latest main too.

https://github.com/Expensify/App/assets/161821005/dd0df08e-048d-40d2-9fe2-29720f72154f

**Comment by ikevin127** [comment]
**`Overdue` notes:** Issue is still reproducible and we're awaiting for the BE part to be completed before we can move forward with FE pull request review.

**Comment by techievivek** [comment]
Okay, I see, thanks. I didn't wait long enough to see if the profile info would disappear. I am re-triaging it now.

**Comment by techievivek** [comment]
Sorry for the delay, last week I was handling fires and stuff so couldn't come to this GH.

**Comment by techievivek** [comment]
@nkdengineer Looking the the solutions you have mentioned.

> Modify the back-end to merge preexistingAccountID into the personalDetailsList item of the current optimistic personal detail created in front end. For example, if the optimisticAccountID created in front-end is 123456, and the actual accountID of the existing user is 654321, then the backend will return this Onyx data:
```
{
  "key": "personalDetailsList",
  "onyxMethod": "merge",
  "value": {
    "123456": {
      "preexistingAccountID": 654321
  }
}
```
Can you get the App PR ready so I can test this with my backend changes? Asking because we don't send the optimistic accountID in the request so the response you are expecting is not possible, right?

**Comment by nkdengineer** [comment]
@techievivek you can test with this open [PR](https://github.com/Expensify/App/pull/42251)

**Comment by techievivek** [comment]
@nkdengineer Added comment on the PR, thanks.

**Comment by ikevin127** [comment]
The PR has been on HOLD for BE PR to be merged for the last 2 weeks.

> The [backend PR](https://github.com/Expensify/Web-Expensify/pull/42279) is being reviewed, I will update here once we merged and deployed it to staging.

**Comment by techievivek** [comment]
This is a tricky one, and @marcaaron is the assigned reviewer for backend changes. It will be great if we can have the discussion here or on slack so we can reach to a agreed solution.

**Comment by techievivek** [comment]
Marc's comment on the PR
> I don't quite understand the premise of this change. It's a single client who generated the accountID. Can they remove it from the Onyx data by applying successData?

My response: 
When we invite a member to the workspace, we ensure that their personal details exist in Onyx. This can happen in one of three ways:

1. The personal details already exist.
2. The personal details are added to Onyx as part of the SearchForReports call, which is used when an admin tries to invite someone.
3. For a non-existing user, if the backend doesn't return any data, we create optimistic data in the frontend.
This optimistic entry becomes the source of truth for the invited user's personal details. Consequently, if these details are deleted in successdata, nothing appears in the profile, which is the bug we're addressing. You can check the screen recordings on GH to see this in action.

Our solution is to retain the optimistic personal details and have the backend inform us of the correct accountID personal details to use.

For example, if the optimistic accountID created in the frontend is 111111111, and the actual accountID of the existing user is 9999999, the backend will return this Onyx data:

```
{
  "key": "personalDetailsList",
  "onyxMethod": "merge",
  "value": {
    "111111111": {
      "preexistingAccountID": 9999999
    }
  }
}
```

Thus, anytime we look up the personal details of 111111111, we will check if there is a key called 9999999 and use the preexistingAccountID personal details if it exists.

I would suggest you take a look the RCA [here](https://github.com/Expensify/App/issues/38925#issuecomment-2053605588) to understand this more.

**Comment by techievivek** [comment]
There are some subsequent comments after my above reply. I am going to address them here.

**Comment by techievivek** [comment]
> > For a non-existing user, if the backend doesn't return any data, we create optimistic data in the frontend.

> Why is the backend not returning any data?

Backend is returning the data but the accountID generated from backend is different from optimistic data generated in frontend.
--------------------------

> > This optimistic entry becomes the source of truth for the invited user's personal details.

> I'm not sure what you mean by this. Last time I checked, accountID can't be generated by the client for use by the backend.

So when you invite the user, we generate optimistic accountID/personal details for the user, so if you were to navigate to user's profile it would use the optimistic data to show the info and not the returned data from the backend.

------------

>> Consequently, if these details are deleted in successdata, nothing appears in the profile, which is the bug we're addressing. You can check the screen recordings on GH to see this in action.

> Ok, but why are they not getting replaced with actual details of the newly created user? That's why I'm asking if it is safe to remove them from the successData.

Because the returned accountID from backend is different from optimistic accountID that was generated in the frontend and there is no mechanism to merge data as the accountIDs are different.

-------------

>> Our solution is to retain the optimistic personal details and have the backend inform us of the correct accountID personal details to use.

> Why not just have the backend send the correct personal detail? What does the preexistingAccountID field do for us?

`preexistingAccountID` is one way to tell that the optimistic personal details generated is actually matched to a different accountID in backend and it forces it use to the backend optimistic data that was returned.

--------------------

>> Thus, anytime we look up the personal details of 111111111, we will check if there is a key called 9999999 and use the preexistingAccountID personal details if it exists.

> This might be easier to understand if we discussed a specific example. I'm sort of confused because personal details also have logins. So, it feels like this is a possibility:

> We are using an optimistic accountID to do "something" when we should maybe be using a login instead. The server can always know the "accountID". The client can figure out the accountID if they have the personal details by using the login to find the correct one.
We are not sending the correct personal details back to the client when the account is created.
WDYT?

Yeah, we don't know need to explicitly send a new key `preexistingAccountID` for this case, we can definitely handle this with the login as that hasn't changed. But the contributor suggested the `preexistingAccountID` approach since we follow something similar in case of optimistic reports, which they have described in their proposal.

**Comment by techievivek** [comment]
@nkdengineer, can you please review the above discussion and confirm if I have correctly understood the proposed solution? Additionally, could you explain any details I might have missed or provide further clarification to help @marcaaron understand the solution? Thanks.

**Comment by nkdengineer** [comment]
@techievivek Yes, you're correct. That is exactly what I proposed.

**Comment by marcaaron** [comment]
It feels like we are maybe missing something. The optimistic `accountID` can be thrown away when the request succeeds? This requires no backend changes.

>Because the returned accountID from backend is different from optimistic accountID that was generated in the frontend and there is no mechanism to merge data as the accountIDs are different.

Does the backend set the complete data or incomplete?

>we can definitely handle this with the login, as that hasn't changed. But the contributor suggested the preexistingAccountID approach since we follow something similar in case of optimistic reports, which they have described in their proposal

`accountID` and `reportID` are different situations though. The `accountID` that is optimistic will never become real so it shouldn't ever be a problem and the user doesn't need to know about it.

**Comment by marcaaron** [comment]
>if these details are deleted in successdata, nothing appears in the profile, which is the bug we're addressing

- Could we not let the user take this action?
- Or show a not found page?

This seems a bit edge case. Maybe better to avoid complexity if possible?

**Comment by techievivek** [comment]
> Does the backend set the complete data or incomplete?

Sorry, this is not clear to me what data we are talking about here. Backend returns the personal details of newly added user.

**Comment by techievivek** [comment]
> accountID and reportID are different situations though. The accountID that is optimistic will never become real so it shouldn't ever be a problem and the user doesn't need to know about it.

I agree, but then I am not sure why we even went with generating the optimistic data in the frontend, given the generated account would never match the one created in the backend.

**Comment by techievivek** [comment]
> - Could we not let the user take this action?
> - Or show a not found page?

> This seems a bit edge case. Maybe better to avoid complexity if possible?

Yes, there is [another proposal](https://github.com/Expensify/App/issues/38925#issuecomment-2018220673) as well for this. However, I liked the proposal by @nkdengineer because it allows the user to access the app without any constraints and fixes the problem, even though it's a bit more complicated. But even blocking the action here is not a big deal, so maybe we can keep things simple and just prevent the user to access newly added user's profile page.

**Comment by marcaaron** [comment]
>blocking the action here is not a big deal, so maybe we can keep things simple and just prevent the user to access newly added user's profile page

I agree, I can't think of why we would need to be able to immediately navigate to a profile that will only have a login and no other details. Seems like a low value and uncommon user action to take.

**Comment by ikevin127** [comment]
Given the latest discussion here I think we're looking at re-assigning contributor for this issue to @bernhardoj, given they were first to post a proposal that would fix the issue by not allowing user profile to be opened while optimistic.

The reason I went with @nkdengineer's proposal was because at the moment it looked the best in terms of expected result even though it needed BE changes.

##

Here's the difference between the 2 proposals from a behaviour / looks POV we have **bernhardoj**'s [here](https://github.com/Expensify/App/issues/38925#issuecomment-2059096784) and **nkdengineer**'s [here](https://github.com/Expensify/App/issues/38925#issuecomment-2064559658).

##

Before assignment though, if everyone is in agreement, I'd like to get @Expensify/design's take on the looks when it comes to **bernhardoj**'s solution, maybe we can gray-out the option (optimistic UI) or something else more intuitive to the user to let them know that something is loading / happening and that's why the click doesn't seem to do anything, since that might be seen as broken functionality.

**Comment by dannymcclain** [comment]
I kinda like the idea of using the 50% opacity pattern (offline/optimistic/disabled). We could also (or instead?) change the cursor to be the `not allowed` cursor when hovering over the row? Curious what the other designers are thinkin'

**Comment by shawnborton** [comment]
I'd rather not make the row look disabled personally, and I'm not sure if we even need the disabled cursor if we just use a regular cursor instead of the pointer cursor. There's a lot of things in the app you can't interact with and instead of doing a not allowed cursor we just use a regular mouse arrow, right? 

That being said though, I don't feel great about this kind of exception we're creating. Why wouldn't we just be able to allow the user to open up the profile view, and keep the same avatar + login that was used during the invite process? That feels like a much better experience personally. 

cc @puneetlath as I think I remember you chiming in on a previous discussion about this sort of thing where we display user information a certain way from Search/Find but it might change to hidden while chatting. This seems to be somewhat related.

**Comment by ikevin127** [comment]
Based on the design team input I'm a bit confused regarding the **expected result** given the current constraints.
As mentioned above in https://github.com/Expensify/App/issues/38925#issuecomment-2195457143, this is what the two currently existing proposals offer:
> Here's the difference between the 2 proposals from a behaviour / looks POV we have **bernhardoj**'s [here](https://github.com/Expensify/App/issues/38925#issuecomment-2059096784) and **nkdengineer**'s [here](https://github.com/Expensify/App/issues/38925#issuecomment-2064559658).

Based on https://github.com/Expensify/App/issues/38925#issuecomment-2196247877 and https://github.com/Expensify/App/issues/38925#issuecomment-2197032120:

1.
> That being said though, I don't feel great about this kind of exception we're creating. Why wouldn't we just be able to allow the user to open up the profile view, and keep the same avatar + login that was used during the invite process? That feels like a much better experience personally.

2.
> I agree this would be the better user experience, just wasn't sure if there was some technical limitation that was absolutely preventing us from doing it this way.

It seems like we're looking for an alternative solution (if possible) which expects a slightly different behaviour then what the current proposals offer.

## Summary
- **nkdengineer** already invested some time in opening a PR with their solution which would fulfil the issue's **expected result** but required BE changes
- **marcaaron** suggested to avoid BE changes due to complexity and that being able to immediately navigate to a profile that will only have a login and no other details seems like a low value and uncommon user action
- **dannymcclain** (design team) mentioned initially that the grayed-out offline pattern would look good for if we were to go with **bernhardoj**'s solution which suggested not allowing to open user profile RHP on row click
- further discussions between design team **shawnborton** and **dannymcclain** suggested a possible alternative solution where we still open user profile RHP and keep the same avatar + login that was used during the invite process (basically blocking the change when the real BE data comes in ? not sure)

cc @techievivek As I'm not sure where should we go from here and what should I request from the contributors (if still interested and if any changes are required due to the possible updated **expected result**).

**Comment by puneetlath** [comment]
Hmm, yeah this is a challenge because we use an optimistic accountID originally and then the back-end returns the real one. I can't really think of a straightforward solution either other than disabling the row while we wait for the API response.

**Comment by techievivek** [comment]
Yeah, I think we can just move with disabling the row for now. @bernhardoj mentioned in [their proposal](https://github.com/Expensify/App/issues/38925#issuecomment-2018220673) that we do this in other flows as well.

**Comment by techievivek** [comment]
We can discuss the compensation for @nkdengineer once the issue is fixed. CC @strepanier03. We had to reassign the issue to @bernhardoj, but @nkdengineer had already pushed a PR which we decided to close since it involved complex changes in backend.

**Comment by bernhardoj** [comment]
PR is ready

cc: @ikevin127

**Comment by bernhardoj** [comment]
I'll request in ND once payment is due.

**Comment by ikevin127** [comment]
* [x]  [@ikevin127] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/22622
* [x]  [@ikevin127] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/22622/files#r1676435226
* [x]  [@ikevin127] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **N/A**.
* [x]  [@ikevin127] Determine if we should create a regression test for this bug.
* [x]  [@ikevin127] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

## Regression Test Proposal

**Precondition:** User is admin of a Collect workspace.

1. Go to Profile > Workspaces > Collect workspace.
2. Go to Members.
3. Invite a new member that does not have an Expensify account, using an email address.
4. Immediately after inviting the member, verify that the newly added member row is **disabled** until the BE returns the created user information. While this happens, the default avatar should be shown.

Do we agree 👍 or 👎.

**Comment by nkdengineer** [comment]
> We can discuss the compensation for @nkdengineer once the issue is fixed.

@techievivek Could you help with the discussion [here](https://github.com/Expensify/App/issues/38925#issuecomment-2202573432), thanks 🙇

**Comment by ikevin127** [comment]
@strepanier03 This was deployed to production 5 days ago ([reference](https://github.com/Expensify/App/pull/44734#issuecomment-2220221811)) meaning payment is due on **17th of July**.

**Comment by bernhardoj** [comment]
As @ikevin127 mentioned, the payment is due on the 17, so I've requested it in ND.

**Comment by strepanier03** [comment]
@bernhardoj - I closed the contract in Upwork without payment since you're requesting in Newdot now. 

@ikevin127 - I've paid your contract and closed it as well. 

----

@nkdengineer  and @techievivek - What [payment](https://github.com/Expensify/App/issues/38925#issuecomment-2202573432) should I authorize here? The full $250?

**Comment by strepanier03** [comment]
### Payment summary

- $250 - @bernhardoj via Manual Requests in Newdot (requested)
- $250 - @ikevin127 via Upwork (paid)
- $? - @nkdengineer via Upwork (pending)

@JmillsExpensify - Request incoming.

**Comment by JmillsExpensify** [comment]
$250 approved for @bernhardoj
